src/main/java/com/thebois/models/inventory/Inventory.java

```diff
@@ -0,0 +1,141 @@
+            add(item);
+        }
+    }
+
+    /**
+     * Removes and returns the first item of the specified type.
+     *
+     * @param itemType The item type to be taken.
+     *
+     * @return The item removed from the inventory.
+     *
+     * @throws IllegalArgumentException If the item type is not in the inventory.
+     */
+    public IItem take(final ItemType itemType) {
+
+        if (hasItem(itemType)) {
+            final int firstIndexOfItemType = getFirstIndexOf(itemType);
+            return items.remove(firstIndexOfItemType);
+        }
+        else {
+            throw new IllegalArgumentException("Specified ItemType not in inventory");
+        }
+    }
+
+    /**
+     * Take multiple items from the inventory.
+     *
+     * @param itemType The item type to be taken.
+     * @param amount   The amount of items to be taken.
+     *
+     * @return The list of items removed from the inventory.
+     *
+     * @throws IllegalArgumentException If the amount given is greater that the items of the
+     *                                  specified item type in the inventory.
+     */
+    public ArrayList<IItem> takeAmount(final ItemType itemType, final int amount) {
+        final ArrayList<IItem> taken = new ArrayList<>(amount);
+        for (int i = 0; i < amount; i++) {
+            try {
+                taken.add(take(itemType));
+            }
+            catch (final IllegalArgumentException exception) {
+                throw new IllegalArgumentException(
+                    "Not enough of the specified ItemType in the inventory");
+            }
+        }
+        return taken;
+    }
+
+    private int getFirstIndexOf(final ItemType itemType) {
+        int index = -1;
+        for (final IItem item : items) {
+            if (item.getType().equals(itemType)) {
+                index = items.indexOf(item);
+            }
+        }
+        return index;
+    }
+
+    /**
+     * Check if an item type is in the current inventory.
+     *
+     * @param itemType The item type to be checked for.
+     *
+     * @return A boolean indicating if the item is in the inventory or not.
+     */
+    public boolean hasItem(final ItemType itemType) {
+        for (final IItem item : items) {
+            if (item.getType().equals(itemType)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Check if a given amount of an item type is in the current inventory.
+     *
+     * @param itemType The item type to be checked for.
+     * @param amount   The amount of to be checked for.
+     *
+     * @return A boolean indicating it the amount of the specified item is in the inventory or not.
+     */
+    public boolean hasItem(final ItemType itemType, final int amount) {
+        for (int i = 0; i <= amount; i++) {
+            if (!hasItem(itemType)) {
+                return false;
+            }
+        }
+        return true;
+    }
```

Comment:
This method does not look like it's doing what it says it does.

It repeatedly checks if the inventory has an item, but it never checks that there are *enough* of them. It just keeps checking for the existance of *any amount* of the item n times. 

A way to implement it correctly is to use `numberOf` and make sure it's greater than or equal to the requested amount.

Reply:
Same goes for the ordinary `hasItem` method, you've already implemented the looping-over-items part in `numberOf` so you can just `return numberOf(itemType) >= 1;`

Reply:
Whoops

Reply:
Anyways, this should've been caught by *the tests* that you've definitely got for these methods, right? 👀

Reply:
False positives, though I thought I checked it ☹️

Reply:
That's interesting, now I'm curious, were you missing a specific test case or was it a structural mistake in the test that made it miss this error?

Reply:
Went back and checked if it would give a false positive now after you comments, and it did. And I didn't commit what I tested since I just tested it wouldn't give me a false positive so I don't remember exactly _how_ I tested it ☹️

Reply:
Ah, okay. Well, these edge cases are really easy to miss when writing tests so good thing we have code reviews 😄

Reply:
💯